server/src/test/YmlParsingErrorListener.test.ts

```diff
@@ -0,0 +1,29 @@
+import { CharStreams, CommonTokenStream } from 'antlr4ts';
+import { Diagnostic } from 'vscode-languageserver';
+
+import { YmlLexer, YmlParser } from '../grammar';
+import { YmlParsingErrorListener } from '../visitors';
+
+function getDiagnostics(text: string): Diagnostic[] {
+    const diagnostics: Diagnostic[] = [];
+    const inputStream = CharStreams.fromString(text, '/path/to/file');
+    const lexer = new YmlLexer(inputStream);
+    const tokenStream = new CommonTokenStream(lexer);
+    const parser = new YmlParser(tokenStream);
+    parser.removeErrorListeners();
+    parser.addErrorListener(new YmlParsingErrorListener(diagnostics));
```

Comment:
Does that constructor add stuff into `diagnostics`?

Reply:
Yes, it does. That's what is tested below.

Reply:
Ho. Sorry. The constructor doesn't add stuff into `diagnostics` but when an error is found, the `listener` adds stuff to `diagnostics`

Reply:
Hum. Because when one reads that function too fast, they may wonder why we carry an empty array around.

Reply:
I added a comment. See 64c9f9b.